crates/aptos-rosetta/src/construction.rs

```diff
@@ -264,25 +264,20 @@ async fn construction_metadata(
         max_gas.0
     } else {
         let account_balance = rest_client
-            .get_account_balance(address)
+            .get_account_balance_bcs(address, &native_coin_tag().to_string())
             .await
             .map_err(|err| ApiError::GasEstimationFailed(Some(err.to_string())))?
             .into_inner();
 
         let maximum_possible_gas =
             if let InternalOperation::Transfer(ref transfer) = request.options.internal_operation {
                 std::cmp::min(
-                    (account_balance
-                        .coin
-                        .value
-                        .0
-                        .saturating_sub(transfer.amount.0))
-                        / gas_price_per_unit,
+                    (account_balance.saturating_sub(transfer.amount.0)) / gas_price_per_unit,
```

Comment:
What if gas_price_per_unit is 0 like in tests?

Reply:
That's handled by the estimation API, it always returns 1 as the lowest.  In theory, if I had a better way to get the min possible gas price, it could be 0.